commands/serve/lib/oauth-router.js

```diff
@@ -0,0 +1,121 @@
+const express = require('express');
+const { Auth, AuthType } = require('@qlik/sdk');
+
+let prevHost = null;
+let prevClientId = null;
+let authInstance = null;
+
+const getAuthInstance = (returnToOrigin, host, clientId) => {
+  if (authInstance && prevHost === host && prevClientId === clientId) {
+    return authInstance;
+  }
+
+  prevHost = host;
+  prevClientId = clientId;
+
+  authInstance = new Auth({
+    authType: AuthType.OAuth2,
+    host,
+    clientId,
+    redirectUri: `${returnToOrigin}/auth/login/callback`,
+  });
+  return authInstance;
+};
+
+const OAuthRouter = () => {
+  const router = express.Router();
+
+  let cachedHost = null;
+  let cachedClientId = null;
+
+  router.get('/oauth', async (req, res) => {
+    const { host: qHost, clientId: qClientId } = req.query;
+    if (!cachedHost && !cachedClientId) {
+      cachedHost = qHost;
+      cachedClientId = qClientId;
+    }
+
+    const returnTo = `${req.protocol}://${req.get('host')}`;
+    const instacne = getAuthInstance(returnTo, qHost, qClientId);
+    const isAuthorized = await instacne.isAuthorized();
+    if (!isAuthorized) {
+      const { url: redirectUrl } = await instacne.generateAuthorizationUrl();
+      res.status(200).json({ redirectUrl });
+    } else {
+      const reqHost = req.get('host');
+      const redirectUrl = `${req.protocol}://${reqHost}/app-list?engine_url=wss://${cachedHost}&qlik-client-id=${cachedClientId}&shouldFetchAppList=true`;
+
+      cachedHost = null;
+      cachedClientId = null;
+
+      res.redirect(redirectUrl);
+    }
+  });
+
+  router.get('/login/callback', async (req, res) => {
```

Comment:
I get that the redirect callback now needs to be http://localhost:8000/auth/login/callback - was that intended? Could we simplify it and document it?

Reply:
I was actually had that in top of my head (since i saw it in [your PR](https://github.com/qlik-oss/nebula.js/blob/6c4cd87c5b23a0dca188ce88f7b900a8dcdc42d6/commands/serve/web/components/Hub/SelectEngine/ConnectionGuid.jsx#L54) as well) to find it update related documentation about that callback url change.

I came across [this one](https://qlik.dev/authenticate/oauth/create-oauth-client-spa/)  but it's not in context of nebula. 

probably we need to update the same link that you had in your pr, hinting about callback url now has the /auth/ part in it. 

There is also an alternate solution of having a `/login/callback` endpoint in root while it redirects the req to `/auth/login/callback`, but not recommending it!